src/app/core/services/user.service.ts

```diff
@@ -33,11 +32,11 @@ export class UserService {
         this.currentUser = this.createUser(jsonData, userLoginId.toLowerCase());
         return this.currentUser;
       }),
-      flatMap((user) => {
+      map((user) => {
```

Comment:
Instead of using a `map`, should we use a `filter` and [throwIfEmpty](https://rxjs.dev/api/operators/throwIfEmpty) ?
The `filter` will validate the `user` object.

Alternatively, we can also simplify the `map` operation as such:
```
map((user) => {
  if (!user) {
    throw new Error('Unauthorized user.');
  }
  return user;
}
```

Reply:
Thanks for the great suggestion! Done!